### DISCUSSION TITLE
Industry Research

### DISCUSSION THREAD

**Discussion by nebhead** [nebhead/PiFire#60] [Ideas]
Currently watching this great video on YouTube from Mad Scientist BBQ which gets into the weeds on Pellet Grill Controllers.  I found it really interesting to watch:  https://youtu.be/CmCibXcfmqY  

One idea I've gleaned from this is to look at adding a new fan (DC - computer fan with PWM?) to add some more options for modulating temperature. 

Enjoy!

**Comment by faisalhusain007** [comment]
Hey Ben, been a long time since I checked in. I am finally back in sunny KL, been away for almost a year, just catching up with all the latest developments. Kudos to you and all the community here with the development of this project. I watched the Vid, I do actually own the 12v GMG but haven't used it since upgrading to my frankenstein traeger bbq75 thanks to you and PiFire.

I  am also interested in going the variable speed on both the Auger and Fan, do you remember really early on when we were experimenting with the SSR we managed to use PWM to change the speed of the fan our goal was to get a better smoke flavour, but we decided to pause the  fan at timed intervals as an easier solution. I think I posted it here https://github.com/nebhead/PiFire/discussions/15#discussioncomment-256520.

 Anyways I have finally tidied up my install and my grill looks kind of normal now.

https://user-images.githubusercontent.com/36872476/147724991-a8a5e466-80a4-4b45-b764-ff97f91fc926.mp4

**Reply by nebhead** [reply]
> Hey Ben, been a long time since I checked in. I am finally back in sunny KL, been away for almost a year, just catching up with all the latest developments. 

Welcome back @faisalhusain007!!  Awesome to see your setup, and it looks great!  I like the clean box look.  

> I am also interested in going the variable speed on both the Auger and Fan, do you remember really early on when we were experimenting with the SSR we managed to use PWM to change the speed of the fan our goal was to get a better smoke flavour, but we decided to pause the fan at timed intervals as an easier solution. I think I posted it here [#15 (comment)](https://github.com/nebhead/PiFire/discussions/15#discussioncomment-256520).

Yep, definitely remember that early on.  I think we can revisit this now, as it seems like it might be a good area to start optimizing even further.  @weberbox just committed some changes to the PID which I think make it even better than it was before, but there is always room for more improvement.  For some of us that have the SSRs, I think we can look into the PWM switching for fans, but for anyone with mechanical relays, they would be out of luck - unless they had a DC fan they could hook up.  I think it's worth looking into that option too.  I also really like the idea of being able to run the whole thing from 12V DC.  A future board revision might be in order.  :) 

Interesting to hear in the video how other companies are achieving their set temperatures, and in some cases obscuring the true temperature swings.  I know most of us would rather see the actual temperatures, I think it's interesting to have an 'easy' mode that does more averaging, or shows the user the rough set temperature.  

Anyway, good to have you back, looking forward to what 2022 will bring for this project!

**Comment by weberbox** [comment]
I was looking up what fan Treager was using as I was thinking a pc fan might not have enough cfm. 

The Treager fan is like $90 which is insane so I researched the specs and found the info below

[Specs](https://de.farnell.com/en-DE/nmb-technologies/11925sa-12r-eu-d1/axial-fan-0-62a-7-44w-12v-119x25mm/dp/3583421)

Seems to be a 120mm fan with 128cfm 

I found a few alternatives on Mouser and this one looks pretty good and much cheaper at $19

[Mouser Fan](https://www.mouser.com/ProductDetail/CUI-Devices/CFM-A225-13-22?qs=2KpglrJGpg8nMlBPB%252BDhHQ%3D%3D)

**Reply by faisalhusain007** [reply]
Hey @weberbox, thanks for all your input in this project, I have been playing with your android app,  great work. I think I also want to go DC fan route, seems too much effort to use AC fan control. In previous projects I use a MOSFET to control fan speeds using PWM. I use these modules because they are value for money 
![image](https://user-images.githubusercontent.com/36872476/147941998-3d775a0d-c8b0-4ce2-bd97-e638e17c07c3.png)

Wondering what you guys think?

**Reply by nebhead** [reply]
Was away a couple days myself, but am back now.  I think @weberbox has the right idea.  And signalling with a MOSFET could definitely work for us.  I think it would be easiest to design a new PCB with this integrated.  But we also will need a supply with 12V as well.  I think @weberbox had suggested it might be good to just build a supply into the design so that it's all fairly compact.  I'm not experienced with power supply design so I might leave that up to him. :) Otherwise we could source something like this: https://www.amazon.com/MEAN-WELL-RD-35A-Supply-Output/dp/B00FRBN2CK/  which has both rails already.  If we are going all in with 12V DC components, then maybe look at what could be done to replace both the auger motor and the igniter as well.

**Reply by nebhead** [reply]
Oops... I think that power supply can only select 12V or 5V.   Maybe something like this then: https://www.amazon.com/Arcity-Switching-Adjustable-Machine-Cocktail/dp/B07RT54H9V/

**Reply by weberbox** [reply]
I think there are a couple options here but they would all need research. If the plan is to swap out all 120v components with 12v then first we would need to know the required amperage of everything and design or find a psu that can supply it. 

I noticed in a few pictures that Treager is just using a power brick like a laptop. While I hate power bricks it could be beneficial to use one. This would allow the grill to be run off a 12v battery when camping etc. Just have a 12v port on the grill and either use the power brick or alligator clip type cord to a battery. That could also be done even if the psu is built into the board I guess. 

Also I do not think there is a need to get a PSU with 12v and 5v output. Just 12v then have a simple step down circuit on the PiFire board to supply the 5v to the Pi etc that should make finding a psu easier. 

I just did a quick search and it seems the Treager bricks are 20v so I wonder if the igniter requires more than 12v

**Reply by faisalhusain007** [reply]
My GMG comes with a 12v Brick, from experience I think the igniters are going to require the most amount of current, followed by the auger lastly the fan. I feel there would be no benefit in going DC on the whole system especially the cost involved in swapping out the components. I think the manufactures only switched to DC to save production costs, for different regions, I had a nightmare setting up my grill here, it is designed to run on US 110/120 VAC @ 60Hz here we have 220/240 VAC @ 50Hz. Ordinarily you can use a step down transformer, however for inductive loads frequency is important, can cause overheating and also in the case of motors they will run at different speeds etc. Going DC with a switch mode power supply means you don't need different components for different regions. 
 In our case we can design a switch mode power supply (12V) and as @weberbox suggested  just use a simple step down circuit (for the 5V) or even a small step down buck if we want better efficiency than a voltage regulator (7805).  

I think  having control  over the speed of the fan is key to getting that perfect smoke.

**Reply by nebhead** [reply]
So, would something like [this](https://smile.amazon.com/SUPERNIGHT-TM-100V-Switching-Adapter/dp/B008FKDK2M) be what you are thinking of?  Agreed, we'd need to find something that has enough current to support running the auger, fan, igniter and raspberry pi + components with a little headroom.  As you said, maybe it's easier to only consider switching the fan to DC, given the other components are readily available and cheap.

**Reply by nebhead** [reply]
Also... just found this fan which is PWM 5V: https://www.amazon.com/Noctua-NF-A12x25-5V-Premium-Adaptor/dp/B07DXLCVVL  

Not sure if it's got the power to push the same CFM as the Traeger fan, but it's a thought.  And if it can work with 3V signaling on the PWM we might not even need extra circuitry.

**Reply by weberbox** [reply]
> Also... just found this fan which is PWM 5V: https://www.amazon.com/Noctua-NF-A12x25-5V-Premium-Adaptor/dp/B07DXLCVVL  
> 
> Not sure if it's got the power to push the same CFM as the Traeger fan, but it's a thought.  And if it can work with 3V signaling on the PWM we might not even need extra circuitry.  

Most Noctu fans are more geared towards silent operation than cfm

That fan is only 60cfm. Maybe we can find a fan that is similar to the 120v one that is used now but 12v or 5v but I think 5v is going to be harder to find
 Might be easier to mount and since the fan blades can be much larger there is a better chance at finding one with the proper cfm

**Reply by weberbox** [reply]
> So, would something like [this](https://smile.amazon.com/SUPERNIGHT-TM-100V-Switching-Adapter/dp/B008FKDK2M) be what you are thinking of?  Agreed, we'd need to find something that has enough current to support running the auger, fan, igniter and raspberry pi + components with a little headroom.  As you said, maybe it's easier to only consider switching the fan to DC, given the other components are readily available and cheap.  

Ya if we go the power brick route something like that. I think the Treager ones were like 4.5 amp but I didn't look very hard

**Reply by faisalhusain007** [reply]
> So, would something like [this](https://smile.amazon.com/SUPERNIGHT-TM-100V-Switching-Adapter/dp/B008FKDK2M) be what you are thinking of? Agreed, we'd need to find something that has enough current to support running the auger, fan, igniter and raspberry pi + components with a little headroom. As you said, maybe it's easier to only consider switching the fan to DC, given the other components are readily available and cheap.

Not quite, I was thinking we could build our own switch mode psu, however given the current required by the ignitor  might be easier to use that brick.

I found this fan 
![image](https://user-images.githubusercontent.com/36872476/148255938-e0ff8e09-22d0-4ce5-85c5-a0d2007eb90b.png)
it uses 12V is PWM with Tacho too

I found this on how to use it with [PWM](https://raspberrypi.stackexchange.com/questions/118486/control-a-12v-dc-fan-via-pwm-and-a-level-shifter) or on a [Arduino](https://fdossena.com/?p=ArduinoFanControl/i.md) Also has RPM detection not sure if that's any use to us but it's there if we need it.

Specs:
We also provide other interfaces for this product, if you need it, please contact me, or in the order remarks,thanks!

Model: QFR1212GHE

Connector: 4 wires

Size: 120*120*38mm

Voltage: DC 12V

Current: 2.70A

Speed: 6200RPM

Air volume: 210.38CFM

Wind pressure: 21.60mmH2O

Noise: 64dB-A

Bearing: High-precision double ball bearing

Fan frame: one-time injection molding, PBT + 30% glass fiber + VO grade flame retardant

Fan blade: one-time injection molding, PBT + 30% glass fiber + VO grade flame retardant.

Fan rotation direction: from the direction of the fan blades, it is counterclockwise

Operating temperature: -10 degrees Celsius to +70 degrees Celsius

Storage temperature: -40 degrees Celsius to +70 degrees Celsius

Power range: plus or minus 15% of the rated power

**Reply by faisalhusain007** [reply]
I got my fan delivered today, didn't get much time to play with it. Just hooked it up to the pifire pcb on IO26 for pwm and ran a test script duty cycles of 75%, 50%, 25% and 0%, using the 3.3v pwm from the pi appears to control the fan speed. I didn't play around with different frequency signals or anything fancy. Hooked it up as is. This fan pushes out too much air 100% duty cycle even at 75% had to use >700g of weight to hold it on my table. 

https://user-images.githubusercontent.com/36872476/148990063-78e4a085-875d-4660-8bc8-14a2c71606da.mp4

**Reply by nebhead** [reply]
Holy moly!  That thing is going to helicopter out of there! :)

**Reply by faisalhusain007** [reply]
Ha! We will anchor to grill use it to move the grill around, add that module to the PiFire, industry first! 
I got it doing the traeger smoke mode thing where it ramps up and down

https://user-images.githubusercontent.com/36872476/149183141-a5f3af11-d8e2-40c3-98d3-620cc860a393.mp4

I have the 12V supply connected to a step down buck powering pi and the fan.  The 12v power supply outputs 12V@2A seems plenty  enough for both the Fan and pi . No other extra components are being used , just using  the 3.3V pwm output from the pi to control the fan. I want to add the pwm stuff to the pid control what do you guys think?

**Reply by nebhead** [reply]
> I have the 12V supply connected to a step down buck powering pi and the fan. The 12v power supply outputs 12V@2A seems plenty enough for both the Fan and pi . No other extra components are being used , just using the 3.3V pwm output from the pi to control the fan. I want to add the pwm stuff to the pid control what do you guys think?

Yeah, maybe create a new grillplat_pifire_dcfan.py module to play with, and see where it takes you.

**Reply by faisalhusain007** [reply]
Sure, which pins to use for PWM and do we want to detect the output from the Fan tacho(RPM), if so which pin ?

I will start adding functions we want PWM linked to  the PID control and for the smoke modes we can use the existing stop and start and do want to add a ramping mode like the traegers have ?

**Reply by nebhead** [reply]
TBH, I hadn't thought through the details. :P   I'd say, use any of the pins on the aux header that aren't being used for buttons right now.  

![image](https://user-images.githubusercontent.com/19876886/149365258-2a7e8e1c-7c51-4277-8349-555687ed22a2.png)

(16, 20, 21 are used)  Might check @weberbox's board too to make sure there aren't any conflicts with assigned GPIOs.  Other than that, go crazy! :)  

Thinking this through a bit, you may want to start a new pid.py as well, that we can utilize for systems with the DC fan control.  So basically you'll want to create two new files and for now you'll need to edit control.py to load those files instead of the defaults.  For the grill_platform you may want to look at switching to GPIOZero to control the GPIOs, and furthermore controlling the PWM output in a simpler fashion. https://gpiozero.readthedocs.io/en/stable/api_output.html#pwmoutputdevice 

Using the above would be easy enough to implement the stop/start, as it could potentially ramp up/down when you just turn it on/off using the current smoke+ mode.

**Comment by faisalhusain007** [comment]
Thanks, I have made some progress got the fan working for both normal /hold  modes. For smoke+ I just use range of duty cycles that ramps up the range  (0.1 seconds) then ramps down again. I made some minor changes to common to add a couple io pins one for the pwm and one tacho. Pointed control to my grillplat, works ok so far using GPIO. Where I am stuck is with the PID control, not to sure how to integrate the fan with the auger . 

Any clues?

**Reply by nebhead** [reply]
> Where I am stuck is with the PID control, not to sure how to integrate the fan with the auger .

You know, that's an excellent question.  It'd be something we will have to play with.  At first blush, I would think that having two separate PID's would be necessary.  But I could be completely wrong on this.  It just seems to me, that there would be different inputs and outputs needed for that - but I'm definitely not any kind of expert on PID.  The fan adjustments can be made more often, I would think, than the auger adjustments.  So this might mean an additional modification to the control script, to add a separate fan PID, (could use the same PID module), that gets evaluated more often.  Maybe every 5-10 seconds.  Yeah, I think this is going to take a little deeper thought and research.

**Reply by faisalhusain007** [reply]
yeah that's where I am stuck, trying to get to grips with it in my head, if we have 2 pids would they conflict with each other,although its way easier to have  2 pids I already have the something ready that I can implement, but in real terms would that make the end cook better? I want something that's as easy as set it and forget about it, but the taste of pitmaster working the grill dedicating hours of their life for that perfect smoke imparted protein flavour.  Going back to the original video you posted at the top of this thread it seems we need that 'variance' to create that true bbq flavour, not a perfect deadline in terms of temperature. A few years back I built a Dry Ager (Drinks fridge, fans, UV lights, Ozone  generator, dehumidifier and humidity generator all running of a ESP12 module) I found I needed to add as much as 30% variance to the set parameters to  achieve amazing results.

**Comment by thomasstrach** [comment]
I used to work with PIDs on the industrial field a lot.Using 2 controllers in a row or as follow up controller is possible but sounds a bit too
much in this case. It’s usually used to control a huge amount of heat that will have a very slow impact on the outcome (pool heating) to avoid a huge overshooting. Means it’s not the right control method here for heating air.
When I installed the PiFire on my pellet smoker it was easily running the 500F quite precise but it was struggling with 180F. So I need to find the correct speed of the fan in combination with the PID settings. Too much air will blow out the fire, too many pellets will cause too much heat. In the end I slowed down the fan speed manually to reach the correct ratio of air and settings in PID mode. It took me about 3 days and 10 pounds of pellets.
It would be helpful for me to set different fan speeds for different temperatures or in relation to the temperature. This will make an easy and smooth low temperature accuracy with nice amount of smoke and will also make it easy to reach high temperature for constant fire to grill flamed salmon or make pizza on a stone plate above the fire at 900 F. This will need some further adapted settings…

In combination with the other findings for the smoke mode it sounds very interesting to work with a controlled fan. 
Might have other positive impacts as well, like a faster start when the fan starts running slow in the beginning and most likely less pellet consumption at all.

**Reply by faisalhusain007** [reply]
Hey @thomasstrach thanks for the info. I have been playing with the pid and fan control and see your point. I too think that it maybe better to have different speeds for different temps as a set point rather than using pid control in this instance. Could you please share your findings in your 3 day experiment. @nebhead perhaps we could add to the webgui a section for fan speed startup and smoke mode ramp range and high temp grill speed to max or something like that. The speed is based on the duty cycle and frequency. I found success with frequency over 1KHz. 

Here is the fan ramp function (Smoke+) in grillplat

```
def FanRamp(self):
        self.pi_pwm.start(0)
        for duty in range(0,50,1):
            self.pi_pwm.ChangeDutyCycle(duty) 
            sleep(0.1)
            sleep(0.05)
    
        for duty in range(50,-1,-1):
            self.pi_pwm.ChangeDutyCycle(duty)
            sleep(0.1)
            sleep(0.05)
```

I only use 50% duty cycle because the fan is on my test bench and not secured anymore it will try and take off!

**Reply by nebhead** [reply]
I think you guys are on the right track here.  I'm thinking it might make sense to build a 'spreadsheet' of values such that we have fan speed targets for different temperature ranges that work well.  We could even have these be configurable in the settings webui.  This way we can tweak them as needed?  Or maybe it should just be a set of fixed values?

**Comment by faisalhusain007** [comment]
Maybe both on the development branch and fixed for the main production,
could be not every grill is the same or pellets used etc

**Reply by nebhead** [reply]
I think for now, everything on the development branch until we can gain some confidence.  Then once it's all looking good, push to main.

**Reply by nebhead** [reply]
One other thing, I think would be really cool, is to build-in pellet burn rate.  Or more simply, auger-on time vs. total runtime.  That could be a nice metric to add to the pelletdb, so that we could track what brands of pellets are burning faster, etc.  Just an idea.

**Reply by faisalhusain007** [reply]
> One other thing, I think would be really cool, is to build-in pellet burn rate. Or more simply, auger-on time vs. total runtime. That could be a nice metric to add to the pelletdb, so that we could track what brands of pellets are burning faster, etc. Just an idea.

Yeah for sure 



> I think for now, everything on the development branch until we can gain some confidence. Then once it's all looking good, push to main.

yep agree

**Comment by faisalhusain007** [comment]
@thomasstrach  Please can you share your fan speeds for different temps would be interested to see how we can implement it. Thanks

**Reply by thomasstrach** [reply]
I did not do any modification on the program with digital speed adjustment.
For me it was the easiest and fastest way to insert an analog rpm limiter in the circuit of the fan.
Sorry, this will not help you a lot.

**Reply by faisalhusain007** [reply]
I just needed your findings i.e. type of fan you used size and max airflow value max rpm value of your fan; then what rpm you used for given temperature range , I can try and write some code to  follow your findings . 

@nebhead just wondering where to pull current temperature of the grill probe which variable to use and from where ?

**Reply by nebhead** [reply]
> @nebhead just wondering where to pull current temperature of the grill probe which variable to use and from where ?

In control.py, in the WorkCycle function, you can pull the current Grill temp from the AvgGT object by calling it's average method i.e. `current_temp = AvgGT.average()`

**Reply by faisalhusain007** [reply]
So I have been trying to get the grill temp from control in to grillplat with almost zero success.  The eventual aim is to use the grill temp to dynamically adjust the speed of the fan based on the grill temp . This code seems to work however thats with static temps that i use in the variable. 

 ```
def FanOn(self):
        WAIT_TIME = 1  # [s] Time to wait between each refresh
        MIN_TEMP = 75
        MAX_TEMP = 260
        FAN_LOW = 10 # Duty cycles starting point from low to high
        FAN_HIGH = 60
        FAN_OFF = 0 
        FAN_MAX = 100 # Max duty cycle for Fan
        setFanSpeed = 0
        temp = float(self.current_temp) # read from control workcycle from init - right now just assigning random values to check func
        if temp < MIN_TEMP:
            setFanSpeed = (FAN_OFF)
            print("Fan OFF") # Uncomment for testing
        # Set fan speed to MAXIMUM if the temperature is above MAX_TEMP
        elif temp > MAX_TEMP:
            setFanSpeed = (FAN_MAX)
            print("Fan MAX") # Uncomment for testing
        # Calculate dynamic fan speed
        else:
            step = (FAN_HIGH - FAN_LOW)/(MAX_TEMP - MIN_TEMP)   
            temp -= MIN_TEMP
            setFanSpeed = (FAN_LOW + ( round(temp) * step ))
            print(FAN_LOW + ( round(temp) * step )) # Uncomment for testing
        return ()
        GPIO.output(self.outpins['fan'], self.RELAY_ON) # Using the relay to power up the fan using a 12v step up buck
        self.pi_pwm.start(setFanSpeed)
```

Once I got it working will tidy up the code and use common for the values.

**Reply by nebhead** [reply]
> So I have been trying to get the grill temp from control in to grillplat with almost zero success. The eventual aim is to use the grill temp to dynamically adjust the speed of the fan based on the grill temp . This code seems to work however thats with static temps that i use in the variable.

OK, here's my suggestion.  To get the temperature from grillplat_[whatever_you_named_it].py, you'll need to import the ReadCurrent() function from common.py.  So at the top of your grillplat file, include the following: 

```
from common import ReadCurrent 
``` 

Then in your code: 
```python
def FanOn(self):
       WAIT_TIME = 1  # [s] Time to wait between each refresh
       MIN_TEMP = 75
       MAX_TEMP = 260
       FAN_LOW = 10 # Duty cycles starting point from low to high
       FAN_HIGH = 60
       FAN_OFF = 0 
       FAN_MAX = 100 # Max duty cycle for Fan
       setFanSpeed = 0

       # *** EDIT **** 
       #  Get the current temps from all probes here
       current_temps = ReadCurrent()
       temp = current_temps[0]  #  Grill probe is return at position 0 of the list 
       # *** END EDIT ****

       if temp < MIN_TEMP:
           setFanSpeed = (FAN_OFF)
           print("Fan OFF") # Uncomment for testing
       # Set fan speed to MAXIMUM if the temperature is above MAX_TEMP
       elif temp > MAX_TEMP:
           setFanSpeed = (FAN_MAX)
           print("Fan MAX") # Uncomment for testing
       # Calculate dynamic fan speed
       else:
           step = (FAN_HIGH - FAN_LOW)/(MAX_TEMP - MIN_TEMP)   
           temp -= MIN_TEMP
           setFanSpeed = (FAN_LOW + ( round(temp) * step ))
           print(FAN_LOW + ( round(temp) * step )) # Uncomment for testing
       return ()
       GPIO.output(self.outpins['fan'], self.RELAY_ON) # Using the relay to power up the fan using a 12v step up buck
       self.pi_pwm.start(setFanSpeed)

```

**Reply by faisalhusain007** [reply]
making progress thanks for that, however the value is returned as a string, and I need to convert it to a integer 

```
Traceback (most recent call last):
  File "control.py", line 1386, in <module>
    WorkCycle('Startup', grill_platform, adc_device, display_device, dist_device)
  File "control.py", line 119, in WorkCycle
    grill_platform.FanOn()
  File "/home/pi/pifire/grillplat_pifire_dcfan1_2.py", line 84, in FanOn
    if temp < MIN_TEMP:
TypeError: '<' not supported between instances of 'str' and 'int'
```

**Reply by nebhead** [reply]
Oops... then convert to an int or float:

```
temp = int(current_temps[0])
```

**Reply by faisalhusain007** [reply]
Thanks, I had a busy day at my day job at the clinic, I just saw this, I ended using float. I have tested the code it seems to work (using the adc prototype) It steps up the duty cycle based on the temperature. Would be great to if we had some real life data on temps vs fan speeds.

When I get a chance I will install the fan to my grill and start testing. Thanks for all your help really appreciate it.

**Reply by nebhead** [reply]
> Thanks, I had a busy day at my day job at the clinic, I just saw this, I ended using float. I have tested the code it seems to work (using the adc prototype) It steps up the duty cycle based on the temperature. Would be great to if we had some temps vs fan speeds.

I know how you feel - pesky day job is always getting in the way of my other projects. :)  

> When I get a chance I will install the fan to my grill and start testing. Thanks for all your help really appreciate it.

Excellent!  I think we are all curious how it will go.  This could guide the future board design.   With all of that said, I did a longer cook over the weekend with @weberbox's latest PID tweaks and it's working extremely well.  Having some additional fan control could really be a great improvement to the system.

**Reply by faisalhusain007** [reply]
> @weberbox's latest PID tweaks and it's working extremely well. Having some additional fan control could really be a great improvement to the system.

Yep did quite a few cooks on @weberbox tweaks and they are working really well. Just a afterthought I haven't implemented the rpm  output of the fan do you think we need it?

**Reply by nebhead** [reply]
> Just a afterthought I haven't implemented the rpm output of the fan do you think we need it?

Um... not really sure.  It might be nice to have the feedback but probably not necessary.

**Reply by faisalhusain007** [reply]
I have been testing the grill with the DC fan varying the speed with temp. So the pros are the grill gets to temp super quick, I  mean shockingly quick, I had to repeat the test 3x to make sure that i hadn't misread something Also get lots of good smoke without using smoke+ . The biggest con is that appears to create huge swings with the temp hold mode, you can't get it to hold it will go up and down, the original fan config was very good at stable temps.
<img width="1139" alt="Screen Shot 2022-02-04 at 4 10 38 AM" src="https://user-images.githubusercontent.com/36872476/152477332-41a12601-1a80-489c-a14a-b385371a3184.png">

**Reply by nebhead** [reply]
I like the fast-startup!  In cold weather (for me right now), I sometimes have to go through a couple of startup cycles before it starts outputting heat.  This would be a big benefit from my point of view.  And for the hold mode, even my grill with the stock fan takes a good amount of time to center on any given temperature.  Sometimes 30m or more.  I think some more testing with perhaps a little less air flow might yield some good results.

**Reply by faisalhusain007** [reply]
I am not sure whats happened ! I connected my AC fan back up to test hold mode, before it was rock solid fluttering at the most 5f. Now its going crazy, I have loaded all the settings back to as before, but now look at the crazy swings even more than the DC fan 

![image](https://user-images.githubusercontent.com/36872476/152686652-56a427d2-3c9b-46a4-84b8-915f8e41d4d8.png)


 I am tearing my hair out !!!
Not sure whats causing this

**Reply by nebhead** [reply]
Hmm... maybe it's something to do with the pellets or the ambient temperature?  Sometimes these kinds of changes can cause big variations in behavior.  It's even possible humidity could have an effect.  Definitely opening the lid, but I assume you kept it closed.  Looking at the graph, it does seem very weird - are you using the main branch of the development branch?

**Reply by faisalhusain007** [reply]
Thing is in KL everyday is almost the same, for this test same batch of pellets were used as I usually use. Lid was kept closed throughout.  It dawned on me, the only thing I have done differently for testing is I cleaned the whole grill before fitting the dc fan I didn't bother to install the drip tray, I thought it's not going to make a difference as long as the baffle is there on top of the firepot should be ok. 

Do you think that would make a big difference?  I am guessing not.

**Reply by nebhead** [reply]
Hmm... it might.  It's a pretty big baffle itself and might be something that tempers the heat fluctuations.  It's possibly worth trying in the same config.

**Reply by faisalhusain007** [reply]
Yep It made a huge difference, here is the same config and setup with the drip tray installed check out the graph now

![image](https://user-images.githubusercontent.com/36872476/152993563-5eed3624-89b7-431c-bbb3-8a8c27811166.png)

**Reply by nebhead** [reply]
Oh much better.  How many degrees of swing is it on average?

**Reply by faisalhusain007** [reply]
Around 3 to 5
